source/server/ssl_context_manager.cc

```diff
@@ -6,23 +6,33 @@
 namespace Envoy {
 namespace Server {
 
-class SslContextManagerStub final : public Envoy::Ssl::ContextManager {
+/**
+ * A stub that provides a SSL context manager capable of reporting on
+ * certificates' data in case there's no TLS implementation built
+ * into Envoy.
+ */
+class SslContextManagerNoTlsStub final : public Envoy::Ssl::ContextManager {
   Ssl::ClientContextSharedPtr
   createSslClientContext(Stats::Scope& /* scope */,
                          const Envoy::Ssl::ClientContextConfig& /* config */) override {
-    throw EnvoyException("SSL is not supported in this configuration");
+    throwException();
```

Comment:
nit: I think you can mark `throwException()` as no return, and then you don't need the return statement on the following line. Same below.